iroha_p2p/src/lib.rs

```diff
@@ -0,0 +1,33 @@
+//! This module provides a network layer for holding of persistent connections
+//! between blockchain nodes. Also, it provides for simple use of encryption
+//! to hide the data flowing between nodes.
+
+use std::io;
+
+use iroha_derive::FromVariant;
+use iroha_error::{derive::Error, error};
+pub use network::{Connect, Network, Post, Received};
+
+mod message;
+mod network;
+mod peer;
+
+/// Error types of this crate.
+#[derive(Debug, FromVariant, Error)]
+pub enum Error {
+    /// Failed to read or write
+    #[error("Failed IO operation")]
+    Network(#[source] io::Error),
```

Comment:
May be just put io instead of network here. Network is kinda useless name inside of network itself

Reply:
I don't like `Io` as it is too short :)

Reply:
Its known abbreviation. Also there always will be a context like: `Error::Io`

Reply:
So it is quite descriptive